staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_vmss.go

```diff
@@ -42,11 +57,13 @@ func (ss *scaleSet) AttachDisk(isManagedDisk bool, diskName, diskURI string, nod
 		return err
 	}
 
-	disks := []compute.DataDisk{}
+	unfilteredDisks := []compute.DataDisk{}
 	if vm.StorageProfile != nil && vm.StorageProfile.DataDisks != nil {
-		disks = make([]compute.DataDisk, len(*vm.StorageProfile.DataDisks))
-		copy(disks, *vm.StorageProfile.DataDisks)
+		unfilteredDisks = make([]compute.DataDisk, len(*vm.StorageProfile.DataDisks))
```

Comment:
what about write like following, then you don't need to copy out to a `unfilteredDisks`
```
filteredDisks = make([]compute.DataDisk, len(*vm.StorageProfile.DataDisks))
filterDetatchingDisks(*vm.StorageProfile.DataDisks, filteredDisks)
```

```
func filterDetatchingDisks( ...) {
  for _, disk := range dataDisks {
      if !*disk.ToBeDetached {
         append disk to filteredDisks
   }
  }
}
```